gnc/sub8_perception/nodes/buoy_2d.py

```diff
@@ -103,9 +135,12 @@ def request_buoy(self, srv):
     def request_buoy3d(self, srv):
         print "Requesting 3d pose"
 
-        if (len(self.observations) > 5) and self.multi_obs is not None:
-            estimated_pose = self.multi_obs.multilaterate(self.observations, self.pose_pairs)
-            self.rviz.draw_sphere(estimated_pose, color=(0.2, 0.8, 0.0, 1.0), scaling=(0.5, 0.5, 0.5), frame='/map')
+        if (len(self.observations[srv.target_name]) > 5) and self.multi_obs is not None:
+            estimated_pose = self.multi_obs.multilaterate(self.observations[srv.target_name], self.pose_pairs[srv.target_name])
```

Comment:
Just nitpicking here, but this is not multilateration. It has a specific meaning related to localization using difference in time of arrival of a signal to multiple receivers.